helm/cluster-cloud-director/templates/vcdcluster.yaml

```diff
@@ -11,24 +11,49 @@ spec:
   org: {{ .org }}
   ovdc: {{ .ovdc }}
   ovdcNetwork: {{ .ovdcNetwork }}
-  parentUid: {{ .parentUid }}
-  useAsManagementCluster: {{ .useAsManagementCluster }}
-  rdeId: {{ .rdeId }}
-  {{- if .storage.createStorageClass }}
+  {{- end }}
+
+  {{- with .Values.cluster }}
+  # Not tested
+  controlPlaneEndpoint:
+    host: {{ .controlPlaneEndpoint.host }}
+    port: {{ .controlPlaneEndpoint.port }}
+
+  # One arm is the old implementation with dnat rule + one arm LB
+  loadBalancer:
+    useOneArm: {{ .loadBalancer.useOneArm }}
+
+  # Persistent storage with Named disks
+  {{- with .defaultStorageClass }}
+  {{- if .enable }}
   defaultStorageClassOptions:
-    vcdStorageProfileName: {{ .storage.vcdStorageProfileName }}
-    k8sStorageClassName: {{ .storage.k8sStorageClassName }}
-    useDeleteReclaimPolicy: {{ .storage.useDeleteReclaimPolicy }}
-    fileSystem: {{ .storage.fileSystem }}
+    vcdStorageProfileName: {{ .vcdStorageProfileName }}
+    k8sStorageClassName: {{ .k8sStorageClassName }}
+    useDeleteReclaimPolicy: {{ .useDeleteReclaimPolicy }}
+    fileSystem: {{ .fileSystem }}
   {{- end }}
   {{- end }}
+
+  # Proxy settings
+  proxyConfig:
+    httpProxy: {{ .proxyConfig.httpProxy }}
+    httpsProxy: {{ .proxyConfig.httpsProxy }}
+    noProxy: {{ .proxyConfig.noProxy }}
+
+  # Not tested - not sure how
+  rdeId: {{ .rdeId }}
+  parentUid: {{ .parentUid }}
+  useAsManagementCluster: {{ .useAsManagementCluster }}
+  {{- end }}
+
+  # Authentication  
   userContext:
-    {{- if .Values.userContext.secretRef.useSecretRef }}
+    {{- if .Values.userContext.secretRef.useSecretRef }}  # I don't trust the upstream prioritization.
```

Comment:
What does the comment mean exactly? If there is a way to provide authentication credentials with a secret, I wouldn't add `.username` and `.password` values.

Reply:
Yes, you can do either creds or a secret. Theoretically the upstream implementation prioritizes the secret if both are set but I didn't observe this behaviour so I added this condition.
The chart defaults to using a secret anyway and that's what we'll use but I left the option to use creds to follow the upstream manifests.